src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ParameterIdentificationRun.cs

```diff
@@ -229,10 +235,14 @@ private IReadOnlyList<SimulationRunResults> runSimulations()
 
       private void updateParameterValues(IReadOnlyList<OptimizedParameterValue> values)
       {
-         if (_parameterIdentification.AllIdentificationParameters.Count != values.Count)
-            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_parameterIdentification.AllIdentificationParameters.Count, values.Count));
+         if (_variableParameters.Count != values.Count)
+            throw new ArgumentException(Error.IdentificationParametersAndValuesDoNotTheSameLength(_variableParameters.Count, values.Count));
 
-         _parameterIdentification.AllIdentificationParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+         //Update variable parameters according to optimized parameter values
+         _variableParameters.Each((x, i) => addParameterValue(x, values[i].Value));
+
+         //Preserve fixed parameter values
+         _fixedParameters.Each((x, i) => addParameterValue(x, x.StartValue));
```

Comment:
This could be optimized in the future: fixed parameters could be set at the begin of simulation to their fixed values in SimModel and considered constant from then.
(or just set values of fixed parameters in the optimized **_ModelCore_** simulation before starting the optimization. Probably even better)